### ISSUE TITLE
quick_error!: trait bound Error not satisfied

### ISSUE THREAD

**Issue by Yamakaky** [brson/error-chain#46]
``` rust
quick_error!(
    #[derive(Debug)]
    pub enum DecodeError {
        BasicOp(val: u16) {
            description("invalid basic opcode")
            display("invalid basic opcode: {:x}", val)
        }
        SpecialOp(val: u16) {
            description("invalid special opcode")
            display("invalid special opcode: {:x}", val)
        }
    }
);
```

Compiler error:

```
src/emulator/cpu.rs:52:47: 52:48 error: the trait bound `types::DecodeError: std::error::Error` is not satisfied [E0277]
src/emulator/cpu.rs:52             Error::DecodeError(ref e) => Some(e),
```

`description()` only add a `impl DecodeError` and not `impl Error for DecodeError`.

**Comment by Yamakaky** [comment]
Also, `quick_error!` doesn't really have a doc.

**Comment by Yamakaky** [comment]
Seems to be the culprit https://brson.github.io/error-chain/src/error_chain/src/quick_error.rs.html#280-324

**Comment by Yamakaky** [comment]
Fixed in upstream, closing for #58.
